pkg/testmachinery/testflow/node/node.go

```diff
@@ -169,6 +147,54 @@ func (n *Node) Status() *tmv1beta1.StepStatus {
 	return status
 }
 
+func (n *Node) isRootNode() bool {
+	return n.inputSource != nil
+}
+
+func (n *Node) GlobalArtifacts() []argov1.Artifact {
```

Comment:
`GlobalArtifacts` is misleading.
We should name something like `generateArtifacts` or `getTaksArtifacts`

Reply:
done